core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedDocServicePlugin.java

```diff
@@ -489,6 +472,15 @@ private static DescriptiveTypeInfo newDescriptiveTypeInfo(
             return descriptiveTypeInfo;
         }
 
+        if (typeSignature.type() == TypeSignatureType.REQUEST_OBJECT) {
```

Comment:
`RequestObjectTypeSignature` could be exposed to users through a custom `DescriptiveTypeInfoProvider`. They might find it difficult to interpret `TypeSignatureType.REQUEST_OBJECT`. Because `AnnotatedValueResolversWrapper` is an internal class that users can't access and `RequestObjectTypeSignature` is an internal API.

I guess you don't want users to allow accessing `RequestObjectTypeSignature` since you located it in the internal package. If so, we should also hide `REQUEST_OBJECT` from the prospect of users' view? We may set the `TypeSignatureType` to `TypeSignatureType.STRUCT` and extract `AnnotatedValueResolversWrapper` if the type is `RequestObjectTypeSignature`.
```suggestion
        if (typeSignature instanceof RequestObjectTypeSignature) {
```